### ISSUE TITLE
Pymssql 2.2.8 is not working in AWS GLue

### ISSUE THREAD

**Issue by DmitryGrb** [pymssql/pymssql#833]
**Describe the bug**
Pymssql 2.2.8 is not working in AWS GLue

**To Reproduce**
1. Create glue pyspark job
2. set --additional-python-modules pymssql

**Expected behavior**
Job run fine

**Current Behavior**
`× Building wheel for pymssql (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [43 lines of output]
      setup.py: platform.system() => Linux
      setup.py: platform.architecture() => ('64bit', '')
      setup.py: platform.libc_ver() => ('glibc', '2.26')
      setup.py: include_dirs => []
      setup.py: library_dirs => []
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-39
      creating build/lib.linux-x86_64-cpython-39/pymssql
      copying src/pymssql/__init__.py -> build/lib.linux-x86_64-cpython-39/pymssql
      running build_ext
      warning: src/pymssql/_mssql.pyx:27:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:28:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:29:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:30:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:31:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:32:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:35:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:36:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:38:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:82:0: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:230:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:264:4: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:333:4: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:831:8: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:919:8: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:1005:12: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:1379:16: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      warning: src/pymssql/_mssql.pyx:1713:8: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      Compiling src/pymssql/_mssql.pyx because it changed.
      [1/1] Cythonizing src/pymssql/_mssql.pyx
      building 'pymssql._mssql' extension
      creating build/temp.linux-x86_64-cpython-39
      creating build/temp.linux-x86_64-cpython-39/src
      creating build/temp.linux-x86_64-cpython-39/src/pymssql
      gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/.pyenv/versions/3.9.10/envs/python39_loaded/include -I/.pyenv/versions/3.9.10/include/python3.9 -c src/pymssql/_mssql.c -o build/temp.linux-x86_64-cpython-39/src/pymssql/_mssql.o -DMSDBLIB
      src/pymssql/_mssql.c:1127:10: fatal error: sqlfront.h: No such file or directory
       #include "sqlfront.h"
                ^~~~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]`

**Context (Environment)**
 - Glue 3.0. Python 3. 
 - please include the output of the following command:
    Cannot do this on glue host

**Additional context**
Seems like version 2.2.8 cannot be build under AWS Glue. 2.2.7 works fine

**Comment by yxl1108** [comment]
+1

**Comment by may-fernandes** [comment]
+1

**Comment by ZWMiller** [comment]
+1 for a databricks environment and ubuntu development environment. 

```bash
Collecting pymssql
  Downloading pymssql-2.2.8.tar.gz (171 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 171.5/171.5 kB 3.4 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Building wheels for collected packages: pymssql
  Building wheel for pymssql (pyproject.toml): started
  Building wheel for pymssql (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
  
  × Building wheel for pymssql (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [25 lines of output]
      setup.py: platform.system() => Linux
      setup.py: platform.architecture() => ('64bit', 'ELF')
      setup.py: platform.libc_ver() => ('glibc', '2.35')
      setup.py: include_dirs => []
      setup.py: library_dirs => []
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-310
      creating build/lib.linux-x86_64-cpython-310/pymssql
      copying src/pymssql/__init__.py -> build/lib.linux-x86_64-cpython-310/pymssql
      running build_ext
      cythoning src/pymssql/_mssql.pyx to src/pymssql/_mssql.c
      cythoning src/pymssql/_pymssql.pyx to src/pymssql/_pymssql.c
      building 'pymssql._mssql' extension
      creating build/temp.linux-x86_64-cpython-310
      creating build/temp.linux-x86_64-cpython-310/src
      creating build/temp.linux-x86_64-cpython-310/src/pymssql
      x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/local_disk0/.ephemeral_nfs/envs/pythonEnv-f5c99be0-9861-44aa-8698-673606e44eb4/include -I/usr/include/python3.10 -c src/pymssql/_mssql.c -o build/temp.linux-x86_64-cpython-310/src/pymssql/_mssql.o -DMSDBLIB
      src/pymssql/_mssql.c:747:10: fatal error: sqlfront.h: No such file or directory
        747 | #include "sqlfront.h"
            |          ^~~~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pymssql
Failed to build pymssql
ERROR: Could not build wheels for pymssql, which is required to install pyproject.toml-based projects
```

**Comment by RGAlexander216** [comment]
Same issue in Azure Databricks

**Comment by dwpatrick** [comment]
+1

**Comment by elrehm** [comment]
Yep, seeing something similar this morning when our Pipfile tried pulling in 2.2.28.  We had to revert to 2.2.27 for now:

An error occurred while installing pymssql==2.2.8 --hash=sha256:30bfd7b8edef78097ccd3f52ac3f3a5c3cf0019f8a280f306cacbbb165caaf63 --hash=sha256:9baefbfbd07d0142756e2dfcaa804154361ac5806ab9381350aad4e780c3033e --hash=sha256:381d8a47c4665d99f114849bed23bcba1922c9d005accc3ac19cee8a1d3522dc --hash=sha256:d873e553374d5b1c57fe1c43bb75e3bcc2920678db1ef26f6bfed396c7d21b30 --hash=sha256:e920d6f805a525f19e770e48326a5f96b83d7b8dfd093f5b7015b54ef84bcf4c --hash=sha256:ebe7f64d5278d807f14bea08951e02512bfbc6219fd4d4f15bb45ded885cf3d4 --hash=sha256:895041edd002a2e91d8a4faf0906b6fbfef29d9164bc6beb398421f5927fa40e! Will try again.
2023-07-31T14:05:45.0537200Z Installing initially failed dependencies...
2023-07-31T14:05:46.9036865Z [pipenv.exceptions.InstallError]: Collecting pymssql==2.2.8 (from -r /tmp/pipenv-4mbjhgmz-requirements/pipenv-e0kw8770-requirement.txt (line 1))
2023-07-31T14:05:46.9037626Z [pipenv.exceptions.InstallError]:   Using cached pymssql-2.2.8.tar.gz (171 kB)
2023-07-31T14:05:46.9038379Z [pipenv.exceptions.InstallError]:   Preparing metadata (setup.py): started
2023-07-31T14:05:46.9039052Z [pipenv.exceptions.InstallError]:   Preparing metadata (setup.py): finished with status 'error'
2023-07-31T14:05:46.9039399Z [pipenv.exceptions.InstallError]: error: subprocess-exited-with-error
2023-07-31T14:05:46.9039609Z [pipenv.exceptions.InstallError]:   
2023-07-31T14:05:46.9039944Z [pipenv.exceptions.InstallError]:   × python setup.py egg_info did not run successfully.
2023-07-31T14:05:46.9040264Z [pipenv.exceptions.InstallError]:   │ exit code: 1
2023-07-31T14:05:46.9040757Z [pipenv.exceptions.InstallError]:   ╰─> [6 lines of output]
2023-07-31T14:05:46.9040986Z [pipenv.exceptions.InstallError]:       Traceback (most recent call last):
2023-07-31T14:05:46.9041201Z [pipenv.exceptions.InstallError]:         File "<string>", line 2, in <module>
2023-07-31T14:05:46.9041559Z [pipenv.exceptions.InstallError]:         File "<pip-setuptools-caller>", line 34, in <module>
2023-07-31T14:05:46.9042032Z [pipenv.exceptions.InstallError]:         File "/tmp/pip-install-ca6vnsl2/pymssql_1bb47a5eadf449f080670afa04a715c9/setup.py", line 45, in <module>
2023-07-31T14:05:46.9042335Z [pipenv.exceptions.InstallError]:           from Cython.Distutils import build_ext as _build_ext
2023-07-31T14:05:46.9042694Z [pipenv.exceptions.InstallError]:       ModuleNotFoundError: No module named 'Cython'
2023-07-31T14:05:46.9042915Z [pipenv.exceptions.InstallError]:       [end of output]

**Comment by s7clarke10** [comment]
I'm having the same issue building on a Redhat 8 environment with Python 3.8.

This has affected tap-mssql which uses pymssql as a database driver.

**Comment by ben-lewington** [comment]
Same issue experience when using Poetry to install the latest version of pymssql in a docker container

**Comment by MarleTangible** [comment]
Windows installations also fail. `pipenv install pymssql` yields `metadata-generation-failed` error.

```
(SampleProject) PS C:\Users\MarleTangible\Documents\Developer\SampleProject> pipenv install pymssql
Loading .env environment variables...
Courtesy Notice: Pipenv found itself running within a virtual environment, so it will automatically use that environment, instead of creating its own for any project. You can set PIPENV_IGNORE_VIRTUALENVS=1 to force pipenv to ignore that environment and create its own instead. You can set PIPENV_VERBOSITY=-1 to suppress this warning.
Installing pymssql...
Error:  An error occurred while installing pymssql!
Error text: Collecting pymssql
  Using cached pymssql-2.2.8.tar.gz (171 kB)
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'error'

  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [15 lines of output]
      Traceback (most recent call last):
        File "C:\Users\MarleTangible\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\pipenv\patched\pip\_vendor\pep517\in_process\_in_process.py", line 
363, in <module>
          main()
        File "C:\Users\MarleTangible\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\pipenv\patched\pip\_vendor\pep517\in_process\_in_process.py", line 
345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "C:\Users\MarleTangible\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\pipenv\patched\pip\_vendor\pep517\in_process\_in_process.py", line 
164, in prepare_metadata_for_build_wheel
          return hook(metadata_directory, config_settings)
        File "C:\Users\MarleTangible\.virtualenvs\SampleProject-OuebegsW\lib\site-packages\setuptools\build_meta.py", line 380, in prepare_metadata_for_build_wheel
          self.run_setup()
        File "C:\Users\MarleTangible\.virtualenvs\SampleProject-OuebegsW\lib\site-packages\setuptools\build_meta.py", line 487, in run_setup
          super(_BuildMetaLegacyBackend,
        File "C:\Users\MarleTangible\.virtualenvs\SampleProject-OuebegsW\lib\site-packages\setuptools\build_meta.py", line 338, in run_setup
          exec(code, locals())
        File "<string>", line 45, in <module>
      ModuleNotFoundError: No module named 'Cython'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

Installation Failed
(SampleProject) PS C:\Users\MarleTangible\Documents\Developer\SampleProject>

```

**Comment by termim** [comment]
Fixed in #803
